core/client/db/src/light.rs

```diff
@@ -792,7 +792,7 @@ pub(crate) mod tests {
 
 		{
 			let tree_route = ::client::blockchain::tree_route(
-				&db,
+				|id| db.header(id)?.ok_or_else(|| client::error::Error::UnknownBlock(format!("Unknown block {:?}", id))),
```

Comment:
Hmm, the display impl of `UnknownBlock` is already `UnknownBlock: {}`. So, maybe just add the id as parameter?